src/Chainweb/Crypto/MerkleLog.hs

```diff
@@ -374,15 +374,15 @@ emptyBody = MerkleLogBody mempty
 mapLogEntries
     :: (forall x . IsMerkleLogEntry u x => x -> b)
     -> MerkleLogEntries u h s
-    -> S.Seq b
+    -> V.Vector b
 mapLogEntries f (MerkleLogBody s) = f <$> s
-mapLogEntries f (h :+: t) = f h S.<| mapLogEntries f t
+mapLogEntries f (h :+: t) = f h `V.cons` mapLogEntries f t
```

Comment:
Here the asymptotics have been damaged; `V.cons` is `O(n)` instead of seq's `O(log n)`. Better would be something like this:

```haskell
mapLogEntries f m = V.concat $ go id m
  where
    go dl (MerkleLogBody s) = dl . (V.map f s :)
    go dl (h :+: t) = go (dl . (V.singleton (f h))) t 
```

i.e. concat chunks of vectors collected via dlist, pushing the `O(n)` step to the end.

Reply:
Yes, this place was the original reason to use `Seq`. However, the number of log entries is expected to be small.

How efficient is `V.concat`? Is it doing a single allocation?